adbnx_adapter/adbnx_adapter/adbnx_controller.py

```diff
@@ -0,0 +1,112 @@
+from .abc import ADBNX_Controller
+
+from arango.graph import Graph as ArangoDBGraph
+from networkx.classes.graph import Graph as NetworkXGraph
+
+try:  # Python +3.8
+    from typing import final
+except ImportError:  # Python 3.6, 3.7
+    from overrides import final
+
+
+class Base_ADBNX_Controller(ADBNX_Controller):
+    def __init__(self):
+        self.nx_graph: NetworkXGraph = None
+        self.nx_map = dict()  # Maps ArangoDB vertex IDs to NetworkX node IDs
+
+        self.adb_graph: ArangoDBGraph = None
+
+        self.adb_map = dict()  # Maps NetworkX node IDs to ArangoDB vertex IDs
+
+    def _prepare_adb_vertex(self, vertex: dict, collection: str):
```

Comment:
Because these methods are no longer internal to the ADBNX_Adapter it could make sense to remove these leading underscores as a matter of convention?

Reply:
(resolving based on slack discussion) 

These will be kept as protected methods, using the `_` (underscore) prefix